src/coreclr/jit/optimizer.cpp

```diff
@@ -4767,18 +4767,41 @@ bool Compiler::optIfConvert(BasicBlock* block)
     }
 #endif
 
-    // Duplicate the input of the assign.
+    // Duplicate the destination of the assign.
     // This will be used as the false result of the select node.
-    GenTree* destination = gtCloneExpr(asgNode->AsOp()->gtOp1);
-    destination->gtFlags &= GTF_EMPTY;
+    assert(asgNode->AsOp()->gtOp1->IsLocal());
+    GenTreeLclVarCommon* destination = asgNode->AsOp()->gtOp1->AsLclVarCommon();
+    GenTree*             falseInput  = gtCloneExpr(destination);
+    falseInput->gtFlags &= GTF_EMPTY;
+
+    // Create a new SSA entry for the false result.
+    {
+        // Get the SSA num of the destination.
+        unsigned lclNum            = destination->GetLclNum();
+        unsigned destinationSsaNum = destination->GetSsaNum();
+        assert(destinationSsaNum != SsaConfig::RESERVED_SSA_NUM);
```

Comment:
> Are we guaranteed that this variable is in SSA?

Is it possible to have a variable that isn't in SSA? I just assumed that wouldn't be possible, given that SSA is usually meant to go across everything. I've not yet seen the assert fire.

Reply:
Yes, the JIT has a max limit on the number of locals it will track (`JitMaxLocalsToTrack`), and various other heuristics/limitations (e.g. it never tracks address exposed variables). See `Compiler::lvaSortByRefCount`.

Reply:
`SsaBuilder::IncludeInSsa` has some additional checks of its own too.

Reply:
ok, I changed the assert to a check and fixed the fgValueNumberTree() too